contrib/mod_sftp/auth.c

```diff
@@ -1149,7 +1149,23 @@ static void incr_auth_attempts(const char *user, cmd_rec *pass_cmd) {
     }
 
     dispatch_cmd_err(pass_cmd);
+
+    /* Note that the mod_auth LOG_CMD_ERR handler for PASS commands will remove
+     * the "mod_auth.orig-user" note as cleanup for future PASS commands, which
+     * may use different user names.  Unfortunately, some event listeners
+     * in mod_ban may expect/require the presence of that note; see
+     * Issue #2009.
+     *
+     * Thus we add the note back here, if necessary, and remove it again
+     * afterward.
+     */
```

Comment:
In particular, see the cleanup happening here, along with its accompanying explanation: https://github.com/proftpd/proftpd/blob/master/modules/mod_auth.c#L438